internal/utils/src/wiremock_utils.rs

```diff
@@ -0,0 +1,186 @@
+    id: usize,
+    jsonrpc: &'a str,
+    method: &'a str,
+    params: StarknetParams,
+}
+
+#[derive(Serialize)]
+pub struct GetBlockWithTx {
+    pub block_id: BlockId,
+}
+
+#[derive(Serialize)]
+pub struct EthGetChainId {
+    pub block_id: BlockId,
+}
+
+impl<'a, StarknetParams> StarknetRpcBaseData<'a, StarknetParams> {
+    pub fn get_block_number(params: StarknetParams) -> Self {
+        Self {
+            id: 1,
+            jsonrpc: "2.0",
+            method: "starknet_blockNumber",
+            params,
+        }
+    }
+
+    pub fn get_block_with_tx(params: StarknetParams) -> Self {
+        Self {
+            id: 1,
+            jsonrpc: "2.0",
+            method: "starknet_getBlockWithTxs",
+            params,
+        }
+    }
+
+    pub fn get_block_with_tx_hashes(params: StarknetParams) -> Self {
+        Self {
+            id: 1,
+            jsonrpc: "2.0",
+            method: "starknet_getBlockWithTxHashes",
+            params,
+        }
+    }
+
+    pub fn call(params: StarknetParams) -> Self {
+        Self {
+            id: 1,
+            jsonrpc: "2.0",
+            method: "starknet_call",
+            params,
+        }
+    }
+}
+
+async fn setup_wiremock() -> String {
+    let mock_server = MockServer::start().await;
+    let empty_vec: Vec<&str> = Vec::new();
+    Mock::given(method("POST"))
+        .and(body_json(StarknetRpcBaseData::get_block_number(&empty_vec)))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_raw(
+                    include_str!("data/starknet_blockNumber.json"),
+                    "application/json",
+                )
+                .append_header("vary", "Accept-Encoding")
+                .append_header("vary", "Origin"),
+        )
+        .mount(&mock_server)
+        .await;
+
+    Mock::given(method("POST"))
+        .and(body_json(StarknetRpcBaseData::get_block_with_tx([
+            BlockId::Tag(BlockTag::Latest),
+        ])))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_raw(
+                    include_str!("data/starknet_getBlockWithTxs.json"),
+                    "application/json",
+                )
+                .append_header("vary", "Accept-Encoding")
+                .append_header("vary", "Origin"),
+        )
+        .mount(&mock_server)
+        .await;
+
+    Mock::given(method("POST"))
+        .and(body_json(StarknetRpcBaseData::get_block_with_tx_hashes([
+            BlockId::Tag(BlockTag::Latest),
+        ])))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_raw(
+                    include_str!("data/starknet_getBlockWithTxHashes.json"),
+                    "application/json",
+                )
+                .append_header("vary", "Accept-Encoding")
+                .append_header("vary", "Origin"),
+        )
+        .mount(&mock_server)
+        .await;
+
+    // Get kakarot contract addess
```

Comment:
To simplify order of merges and so on, could you add a `TODO: Use the latest mapping between Starknet and EVM addresses`